### PR TITLE
Describe all the Liquid drops used on the website

### PR DESCRIPTION
**Pull request by hash4di**

### CODE REVIEW

**Comment by hash4di** [comment]
@Fredar, @Mandaryn finished, please review :)

File: content/reference/drops/availability_drop.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Availability Drop
+
+*Availability Drop* allow you to show rental available terms for booking.
```
Comment (Fredar) [inline content/reference/drops/availability_drop.md]: `allows you to show rental availability map`

File: content/reference/drops/base_drop.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Base Drop
+
+*Base Drop* generates paths to default sections of the website.
```
Comment (Fredar) [inline content/reference/drops/base_drop.md]: `urls and paths` would be more precise

File: content/reference/drops/categories_drop.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+~~~ liquid
+  {% if categories.all.size > 0 %}
+    {% for category in categories.all %}
+  ...
+~~~
+
+Available attributes:
+
+* all - iterate through all categories
+* promoted - itarate through only promoted categories.
```
Comment (Fredar) [inline content/reference/drops/categories_drop.md]: iterate

File: content/reference/drops/category_drop.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Category Drop
+
+Give you access to attributes of `Category`.
```
Comment (Fredar) [inline content/reference/drops/category_drop.md]: gives

File: content/reference/drops/contact_request_drop.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Contact Request Drop
+
+Allows to check and show Contact Form errors.
```
Comment (Fredar) [inline content/reference/drops/contact_request_drop.md:3]: We should decide to either use `*XXX Drop*` at the start of this sentence or not

File: content/reference/drops/context_drop.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# Context Drop
```
Comment (Fredar) [inline content/reference/drops/context_drop.md:1]: I am not sure we should document this drop, think its only for internal usage?

File: content/reference/drops/current_page_drop.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
+# Current Page Drop
+
+Return attributes for current page.
```
Comment (Fredar) [inline content/reference/drops/current_page_drop.md]: returns

File: content/reference/drops/destination_drop.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
+# Destination Drop
+
+Return attributes for selected `Destination`, for example in search bar.
```
Comment (Fredar) [inline content/reference/drops/destination_drop.md]: returns

File: content/reference/drops/guest_drop.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
+# Guest Drop
+
+Return attributes for selected `Guests` number.
```
Comment (Fredar) [inline content/reference/drops/guest_drop.md]: returns

File: content/reference/drops/index.md [hunk 10]
```diff
@@ -0,0 +1,42 @@
+# Drops
+
+*Drops* let you provide the user with custom functionality. From the user's perspective a drop acts very much like a Hash, though methods are accessed with dot-notation as well as element selection. A drop method cannot be invoked with arguments. Drops are called just-in-time, thus allowing you to easily load objects.
```
Comment (Fredar) [inline content/reference/drops/index.md]: This sounds more like a sentence for an actual web-designer, not bsa-website client.

File: content/reference/drops/locale_drop.md [hunk 11]
```diff
@@ -0,0 +1,16 @@
+# Locale Drop
+
+Return attributes for active locale.
```
Comment (Fredar) [inline content/reference/drops/locale_drop.md]: returns

File: content/reference/drops/locales_drop.md [hunk 12]
```diff
@@ -0,0 +1,15 @@
+# Locales Drop
+
+Return attributes for available locales.
```
Comment (Fredar) [inline content/reference/drops/locales_drop.md]: returns

File: content/reference/drops/location_drop.md [hunk 13]
```diff
@@ -0,0 +1,16 @@
+# Location Drop
+
+Return attributes for rental location.
```
Comment (Fredar) [inline content/reference/drops/location_drop.md]: returns

File: content/reference/drops/menu_drop.md [hunk 14]
```diff
@@ -0,0 +1,13 @@
+# Menu Drop
+
+Return created menu items Array ordered by position.
```
Comment (Fredar) [inline content/reference/drops/menu_drop.md]: returns

File: content/reference/drops/menu_item_drop.md [hunk 15]
```diff
@@ -0,0 +1,17 @@
+# Menu Item Drop
+
+Return attributes of menu items.
```
Comment (Fredar) [inline content/reference/drops/menu_item_drop.md]: returns

File: content/reference/drops/page_drop.md [hunk 16]
```diff
@@ -0,0 +1,17 @@
+# Page Drop
+
+Return attributes of page.
```
Comment (Fredar) [inline content/reference/drops/page_drop.md]: damn those returns

File: content/reference/drops/menus_drop.md [hunk 17]
```diff
@@ -0,0 +1,14 @@
+# Menus Drop
+
+Check does menu is visible, not canceled and receive `MenuDrop`.
```
Comment (Fredar) [inline content/reference/drops/menus_drop.md]: Hmm I guess you meant that it `returns a MenuDrop for visible menus`

File: content/reference/drops/page_locale_drop.md [hunk 18]
```diff
@@ -0,0 +1,13 @@
+# Page Locale Drop
+
+Return url with locale of page.
```
Comment (Fredar) [inline content/reference/drops/page_locale_drop.md]: also here and please check all the drops

File: content/reference/drops/rental_photo_drop.md [hunk 19]
```diff
@@ -0,0 +1,32 @@
...
+    {% endif %}
+  {% endif %}
+~~~
+
+Available attributes:
+
+* description
+* kind
+* to_s
+* description
```
Comment (Fredar) [inline content/reference/drops/rental_photo_drop.md]: doubled

also missing `sizes`

File: content/reference/drops/rentals_tag_drop.md [hunk 20]
```diff
@@ -0,0 +1,23 @@
+# Rentals Tag Drop
+
+Return rentals with attributes for current website.
```
Comment (Fredar) [inline content/reference/drops/rentals_tag_drop.md]: actually it returns rentals_tag attributes

File: content/reference/drops/review_drop.md [hunk 21]
```diff
@@ -0,0 +1,22 @@
+# Review Drop
+
+Generates review drop for rental.
```
Comment (Fredar) [inline content/reference/drops/review_drop.md]: returns review attributes?

File: content/reference/drops/rental_type_drop.md [hunk 22]
```diff
@@ -0,0 +1,22 @@
+# Rental Type Drop
+
+Return attributes for rental type.
```
Comment (Fredar) [inline content/reference/drops/rental_type_drop.md]: `returns information about rental type` would be more accurate

File: content/reference/drops/rentals_drop.md [hunk 23]
```diff
@@ -0,0 +1,15 @@
+# Rentals Drop
+
+Return rentals for current website.
```
Comment (Fredar) [inline content/reference/drops/rentals_drop.md]: it only returns promoted rentals, nto all of them

File: content/reference/drops/time_drop.md [hunk 24]
```diff
@@ -0,0 +1,14 @@
+# Time Drop
+
+Return attributes with time. You have ability to manage time format.
```
Comment (Fredar) [inline content/reference/drops/time_drop.md]: This is a weird drop, named `Time` and it returns dates only
@ZenCocoon Maybe rename the drop? Is it even used?

Comment (Fredar) [inline content/reference/drops/time_drop.md]: Also the ability to alter date format does not come from the drop, but from the filters.

File: content/reference/drops/contact_request_drop.md (hunk 5 above)
Comment (Fredar) [inline content/reference/drops/contact_request_drop.md:3]: @hash4di Most of the drops below miss this convention.
Decide to use it everywhere or nowhere.

**Comment by Fredar** [comment]
It is helpful to know what attributes you can call on the objects, but there is no way to tell now what objects are present on what page.
@Mandaryn @Azdaroth Maybe this PR should compile a list of objects available for a given page + the ones that are available everywhere?

File: content/reference/drops/contact_request_drop.md (hunk 5 above)
Comment (hash4di) [inline content/reference/drops/contact_request_drop.md:3]: IMHO we can remove _XXX Drop_ from the start. Line above is header with same content. I think that is enough. @Fredar WDYT?

Comment (Fredar) [inline content/reference/drops/contact_request_drop.md:3]: Sounds fair.

File: content/reference/drops/context_drop.md (hunk 6 above)
Comment (Azdaroth) [inline content/reference/drops/context_drop.md:1]: Well, it's used in very specific usecase, but exposed publicly, I think good to have it documented, even just for consistency.

File: content/reference/drops/context_drop.md [hunk 25]
```diff
@@ -0,0 +1,14 @@
...
+
+Allows to check does requested data loaded properly by `true`/`false` boolean statement.
```
Comment (Azdaroth) [inline content/reference/drops/context_drop.md]: Noy sure about the description, it's rather for checking if response is success or not, it kinds means something similar in this case, maybe it `allows to check if any exception happened during the request`?

File: content/reference/drops/review_drop.md [hunk 26]
```diff
@@ -0,0 +1,22 @@
...
+      <strong>{{ review.name }}</strong>
+    </li>
+  {% endfor %}
+~~~
+
+Available attributes:
+
+* name
+* rating
+* comment
```
Comment (Azdaroth) [inline content/reference/drops/review_drop.md:22]: I think `created_at` is also available here.

File: content/reference/drops/search_drop.md [hunk 27]
```diff
@@ -0,0 +1,20 @@
...
+    <div class="col-sm-4">
+      {% include 'snippets/rental_preview' %}
+    </div>
+  {% endfor %}
+~~~
+
+Available attributes:
+
+* display_type
+* display_url
```
Comment (Azdaroth) [inline content/reference/drops/search_drop.md]: `display_url` is a private method, so it won't probably be available here, but `display_list_url`, `display_photos_url` and `display_map_url` would be, there are the ones defined with meta magic.

File: content/reference/drops/search_form_drop.md [hunk 28]
```diff
@@ -0,0 +1,36 @@
...
+* destinations
+* guests
+* guests_choices
+* min_price_for_stay
+* max_price_for_stay
+* length_of_stay
+* rentals_tags
+* rental_types
+* search_url
+* selected_destinations
```
Comment (Azdaroth) [inline content/reference/drops/search_form_drop.md:43]: I think couple of more here from the delegations, `dislay_type`, `min_price`, etc.

File: content/reference/drops/context_drop.md (hunk 6 above)
Comment (Fredar) [inline content/reference/drops/context_drop.md:1]: @Azdaroth And what is the user usecase? :wink:

Comment (Azdaroth) [inline content/reference/drops/context_drop.md:1]: It exists in one template, so might be worth documenting, just to prove it's no magic.

File: content/reference/drops/twitter_drop.md [hunk 29]
```diff
@@ -0,0 +1,19 @@
...
+    <cite title="{{ site.name }}">@{{ site.social.twitter.user }}</cite>
+  </a>
+~~~
+
+Available attributes:
+
+* get_tweets
+* last_tweet
+* tweets
+* twitter
```
Comment (Azdaroth) [inline content/reference/drops/twitter_drop.md]: It's a private method, provably not available.

File: content/reference/drops/twitter_drop.md [hunk 30]
```diff
@@ -0,0 +1,19 @@
...
+
+~~~ liquid
+  <a href="https://twitter.com/{{ site.social.twitter.user }}" class="external">
+    <cite title="{{ site.name }}">@{{ site.social.twitter.user }}</cite>
+  </a>
+~~~
+
+Available attributes:
+
+* get_tweets
```
Comment (Azdaroth) [inline content/reference/drops/twitter_drop.md]: It's a private method, provably not available.

**Comment by Azdaroth** [comment]
@Fredar Yeah, sound useful, would be nice to know what can be used where.

File: content/reference/drops/review_drop.md (hunk 26 above)
Comment (hash4di) [inline content/reference/drops/review_drop.md:22]: yes, added.

File: content/reference/drops/search_form_drop.md (hunk 28 above)
Comment (hash4di) [inline content/reference/drops/search_form_drop.md:43]: added from [search_form_drop](https://github.com/BookingSync/bsa-website/blob/staging/app/drops/search_form_drop.rb#L2)

File: content/reference/drops/context_drop.md (hunk 6 above)
Comment (hash4di) [inline content/reference/drops/context_drop.md:1]: Ok, so I leave it here.

Comment (Fredar) [inline content/reference/drops/context_drop.md:1]: :+1: Havent spotted that single usage earlier

File: content/reference/drops/time_drop.md (hunk 24 above)
Comment (ZenCocoon) [inline content/reference/drops/time_drop.md]: @Fredar this is used in couple places but see the confusion at first. However, that follows Rails convention here; and we could extend this drop with Time.now for example. Feels legit?

Comment (Fredar) [inline content/reference/drops/time_drop.md]: Ok we named it after `Time.zone`
@hash4di  It has to stay then, but it should only describe `today` and `tomorrow`

Comment (hash4di) [inline content/reference/drops/time_drop.md]: So right now It is okay? In current state it is only `today` and `tomorrow`.

**Comment by hash4di** [comment]
@Fredar, @Azdaroth, @ZenCocoon do You have anything to add/suggest?
I've noticed that it would be good to add new attributes to rentals_tag_drop about `logo`, `color` and `private`. WDYT?

**Comment by Fredar** [comment]
I cant think of any recent drop changes.

**Comment by hash4di** [comment]
I've added two new attributes to rentals_tag_drop: `logo` and `color`. If anyone have something to add/change, please comment. If not, it looks like end of ticket :relaxed:

**Comment by Azdaroth** [comment]
Looks good to me :+1: 
Btw, would be much easier to follow changes when reviewing when each change is in separate commit. Can be squashed before merging, but makes life much easier :).

**Comment by hash4di** [comment]
@Azdaroth it was all the time :wink:. I've squashed yesterday and ask for anything else to be sure.
But thanks for advice, I'll remember to hold this till the end, before merging.

**Comment by Azdaroth** [comment]
@hash4di Ok, all good then ;).

**Comment by hash4di** [comment]
@Fredar do you accept too?

**Comment by Fredar** [comment]
Yes

**Pull request merged**